src/main/java/org/folio/consortia/service/impl/CustomFieldServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package org.folio.consortia.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.consortia.client.CustomFieldsClient;
+import org.folio.consortia.client.OkapiClient;
+import org.folio.consortia.domain.dto.CustomField;
+import org.folio.consortia.service.CustomFieldService;
+import org.folio.spring.FolioExecutionContext;
+import org.folio.spring.exception.NotFoundException;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import java.net.URI;
+
+import static java.lang.String.format;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class CustomFieldServiceImpl implements CustomFieldService {
+  private final FolioExecutionContext folioExecutionContext;
+
+  private final OkapiClient okapiClient;
+  public static final String OKAPI_URL = "http://_";
+  private final CustomFieldsClient customFieldsClient;
+  private static final String MOD_USERS = "mod-users";
+  public static final String QUERY_PATTERN_NAME = "name==%s";
+
+  @Override
+  public void createCustomField(CustomField customField) {
+    Boolean isFieldAlreadyPresent = getCustomFieldByName(customField.getName());
```

Comment:
Better to keep service level contract simple, I mean here is service  that works with object CustomField, so service should provide methods like create, get, delete ...
And client service code should invoke get to check if exists before actual creation.

Also getModuleId() should not be public, it should not be visible in contract. But most of all Spring can not create proxy on not public method to wrap with cache behavior.
We can use not dynamic proxies but it can make things more complicated

Reply:
fixed